### PR TITLE
feat/suggestions enhancement

### PR DESCRIPTION
**Pull request by SavvyShah**
[Main PR](https://github.com/appbaseio/reactivesearch/pull/2242)

### CODE REVIEW

File: stories/SearchBoxWithDocumentSuggestions.vue
Comment (mohdashraf010897) [inline stories/SearchBoxWithDocumentSuggestions.vue:1]: @SavvyShah I believe we could just utilize one file for all type of document suggestions stories.
One recommendation is to use internal props to conditionally leverage slots, refer here - https://github.com/appbaseio/vue-playground/pull/70/files#r1314435927

File: stories/index.stories.js [hunk 1]
```diff
@@ -926,6 +929,13 @@ storiesOf('Search Components/SearchBox', module)
 		components: { BaseSearchBox },
 		template: '<base-search-box :subProps="{ showIcon, showFilter: false}"/>',
 	}))
+	.add('with showDistinctSuggestions', () => ({
```
Comment (mohdashraf010897) [inline stories/index.stories.js:932]: @SavvyShah The order of the stories should be maintained, let's have all the document suggestions stories right after the last added story, which in this case is the [compoundClause story](https://github.com/appbaseio/vue-playground/blob/75de207395a6a51e763d4c8dff022887cbd50e53/stories/index.stories.js#L1239).

**Review by mohdashraf010897 — CHANGES_REQUESTED** [review]
Please refer to the review comments

and loom as shared for web PR

File: stories/SearchBoxWithDocumentSuggestions.vue
Comment (SavvyShah) [reply stories/SearchBoxWithDocumentSuggestions.vue:1]: OK. This is useful. We can conditionally render the `#renderItem` slot. Correct?

**Pull request merged**